python/ray/tune/trainable/function_trainable.py

```diff
@@ -184,6 +186,7 @@ def reset_config(self, new_config):
                 logdir=self._storage.trial_working_directory,
                 driver_ip=None,
                 experiment_name=self._storage.experiment_dir_name,
+                run_id=uuid.uuid4().hex,
```

Comment:
Do we actually need to restore `run_id`? Although the new trial reused the actor, it is still a new run, which should have a new uuid? 

Or do you mean the same trial will pause then restore multiple times before it finishes, so that it need to have the same run_id?

Reply:
Another note, paused Train runs will have a new run uuid every time they resume. This adds a few more entries to the train dashboard, but is not a blocking issue for this niche use case.